src/C-DEngine/C-DCommunication/cdeQueuedSender/cde_QueuedSenderHB.cs

```diff
@@ -7,21 +7,35 @@
 using nsCDEngine.ISM;
 using nsCDEngine.ViewModels;
 using System;
+using System.Collections.Concurrent;
+using System.Linq;
 using System.Threading;
 
 namespace nsCDEngine.Communication
 {
     internal partial class TheQueuedSender
     {
         private readonly object lockStartLock = new ();
+        private Timer _tsmHistoryExpirationTimer = null;
+
         internal void StartHeartBeat()
         {
             if (IsQSenderReadyForHB || !TheBaseAssets.MasterSwitch)
                 return;
             IsQSenderReadyForHB = true;
+
             try
             {
-                MyTSMHistory ??= new TheMirrorCache<TheSentRegistryItem>(TheBaseAssets.MyServiceHostInfo.TO.QSenderDejaSentTime);
+                lock (lockStartLock)
+                {
+                    MyTSMHistory ??= new ConcurrentDictionary<string, TheSentRegistryItem>();
+
+                    var qSenderDejaSentTime = TheBaseAssets.MyServiceHostInfo.TO.QSenderDejaSentTime <= 24 * 60 * 60
+                        ? TheBaseAssets.MyServiceHostInfo.TO.QSenderDejaSentTime * 1000
+                        : 24 * 60 * 60 * 1000;
+
+                    _tsmHistoryExpirationTimer ??= new Timer(MyTsmHistoryRemoveExpired, null, qSenderDejaSentTime, qSenderDejaSentTime);
```

Comment:
Please avoid creating new timers as they are very expensive on single core cpus and slower linux platforms (i.e. Feather with Meadow from Wildernesslabs). If you do need this timer, please create a new config variable that the timer can be turned off to optimize on these platforms

Reply:
Changed this to use the global HB timer for TSM-history expiration checks.